src/one-time-payment/lib/logic.ergo

```diff
@@ -14,16 +14,42 @@
 
 namespace org.accordproject.payment.onetime
 
-contract OneTimeUSDPayment over OneTimeUSDPaymentClause {
-  // Simple Clause
-  clause init(request : InitPaymentRequest) : InitPaymentResponse emits org.accordproject.cicero.runtime.PaymentObligation {
-		emit new org.accordproject.cicero.runtime.PaymentObligation{
+import org.accordproject.cicero.runtime.*
+
+contract OneTimePayment over OneTimePaymentContract {
+  /**
+	 * The initialization logic for this clause immediately notifies the buyer that they have an obligation to make a payment
+	 */
+  clause init(request : InitRequest) : InitResponse emits PaymentObligation {
+	  // enforce (state.status = "UNINTIALIZED") 
+    // else throw "This contract has already been initialized. Current state is " ++ state.status;
+		
+    emit new org.accordproject.cicero.runtime.PaymentObligation{
 			contract: contract.contractId,
 			promisor: contract.buyer,
 			promisee: contract.seller,
 			amount: contract.initialAmount,
-			description: contract.partyA ++ " should pay contract amount to " ++ contract.partyB
+			description: contract.buyer ++ " should pay contract amount to " ++ contract.seller
+		};
+		set state new OneTimePaymentState {
+			stateId: "1",
+			status: "OBLIGATION_EMITTED"
 		};
-    return new InitPaymentResponse{}
+    return new InitResponse{}
   }
+
+  /**
+	 * Update the contract state once we've recieved a payment
+	 */
+  clause paymentReceived(request : PaymentReceivedRequest): PaymentReceivedResponse {
+    enforce (state.status = "OBLIGATION_EMITTED") 
+    else throw "Either a payment obligation hasn't yet been emitted by the contract or payment notification has already been received";
```

Comment:
We should add an issue to support globalization/localization of strings like this.

Reply:
https://github.com/accordproject/ergo/issues/269